cmd/kube-controller-manager/app/options/options.go

```diff
@@ -146,5 +150,9 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
 	fs.Float32Var(&s.KubeAPIQPS, "kube-api-qps", s.KubeAPIQPS, "QPS to use while talking with kubernetes apiserver")
 	fs.IntVar(&s.KubeAPIBurst, "kube-api-burst", s.KubeAPIBurst, "Burst to use while talking with kubernetes apiserver")
+	fs.DurationVar(&s.ClusterDrainTime.Duration, "cluster-drain-time", s.ClusterDrainTime.Duration,
+		"Time which should NodeController take to drain the whole cluster. See also max-node-eviction-period.")
```

Comment:
I suggest documenting these together. Below is a rough draft. I think these knobs are a bit confusing but not too bad. Can you make it clear if the rate limit is between pods or between nodes?


It's desirable to configure how quickly pods are drained when their nodes become NotReady. There are two settings, and node controller will use whichever would lead to a faster eviction rate. --cluster-drain-time lets you specify the amount of time you want node controller to take if it has to drain every pod in the cluster. Set this long enough that a human has time to react if something is going wrong! --cluster-drain-max-delay lets you specify the maximum time to wait between evicting two nodes.